core/src/main/java/io/onetable/iceberg/IcebergCatalogConfig.java

```diff
@@ -27,8 +27,20 @@
 
 @Value
 @Builder
-public class IcebergCatalogConfig {
+public class IcebergCatalogConfig implements io.onetable.client.CatalogConfig {
   @NonNull String catalogImpl;
   @NonNull String catalogName;
   @NonNull @Builder.Default Map<String, String> catalogOptions = Collections.emptyMap();
+
+  public String getCatalogImpl() {
```

Comment:
Similar nitpick on the getters here, lombok is already adding these

Reply:
done.